source/common/common/key_value_store_base.cc

```diff
@@ -71,22 +70,23 @@ bool KeyValueStoreBase::parseContents(absl::string_view contents) {
       return false;
     }
     if (checkForTtl(contents)) {
-      uint64_t ttlInt;
+      uint64_t ttl_int;
       auto token = getToken(contents, error);
-      if (token.has_value() && absl::SimpleAtoi(token.value(), &ttlInt)) {
+      if (token.has_value() && absl::SimpleAtoi(token.value(), &ttl_int)) {
         ttl.emplace(std::chrono::duration_cast<std::chrono::seconds>(
-            std::chrono::time_point<std::chrono::steady_clock>(std::chrono::seconds(ttlInt)) -
+            std::chrono::time_point<std::chrono::steady_clock>(std::chrono::seconds(ttl_int)) -
             time_source_.monotonicTime()));
       } else {
         if (!token.has_value()) {
           ENVOY_LOG(warn, error);
         } else {
-          ENVOY_LOG(warn, "TTL was read from disk, but aoti() failed");
+          ENVOY_LOG(warn, "TTL was read from disk but failed to convert to integer");
         }
       }
     }
-
-    addOrUpdate(key.value(), value.value(), ttl);
+    if (!(ttl.has_value() && ttl <= std::chrono::seconds(0))) {
```

Comment:
nit: personal preference, but I find this harder to read... I think it would be easier to read if it was:

`if (!ttl.has_value() || ttl > std::chrono::seconds(0))`